src/main/java/com/ericsson/ei/controller/FailedNotificationControllerImpl.java

```diff
@@ -60,27 +64,38 @@
     public ResponseEntity<?> getFailedNotifications(
             @RequestParam(value = "subscriptionName", required = true) final String subscriptionName,
             final HttpServletRequest httpRequest) {
-        ObjectMapper mapper = new ObjectMapper();
-        QueryResponse queryResponse = new QueryResponse();
-        QueryResponseEntity queryResponseEntity = new QueryResponseEntity();
-        try {
-            List<String> response = processMissedNotification.processQueryMissedNotification(subscriptionName);
-            if (!response.isEmpty()) {
-                queryResponseEntity = mapper.readValue(response.get(0), QueryResponseEntity.class);
+        Set<String> subscriptionNameList = new HashSet<>(
+                Arrays.asList(subscriptionName.split(",")));
+        JSONArray foundArray = new JSONArray();
+        List<String> notFoundList = new ArrayList<>();
+
+        subscriptionNameList.forEach(name -> {
+            try {
+                List<String> response = processMissedNotification.processQueryFailedNotification(
+                        name);
+                if (!response.isEmpty()) {
+                    JSONObject object = new JSONObject(response.get(0));
+                    foundArray.put(object);
+                    LOGGER.debug("Successfully fetched failed notification for subscription [{}]",
+                            name);
+                }
+            } catch (Exception e) {
+                LOGGER.error("Failed to fetch failed notification for subscription {}", name, e);
+                notFoundList.add(name);
             }
-            queryResponse.setQueryResponseEntity(queryResponseEntity);
-            LOGGER.debug("The response is : {}", response.toString());
-            if (processMissedNotification.deleteMissedNotification(subscriptionName)) {
-                LOGGER.debug("Failed notification for subscription {} was successfully removed from database",
-                        subscriptionName);
-             }
-            return new ResponseEntity<>(queryResponse, HttpStatus.OK);
-        } catch (Exception e) {
-            String errorMessage = "Failed to extract the data from the failed notification object based on subscription name "
-                    + subscriptionName + ".";
-            LOGGER.error(errorMessage, e);
+        });
+        HttpStatus httpStatus;
+        if (foundArray.length() > 0) {
+            httpStatus = HttpStatus.OK;
+        } else {
+            httpStatus = HttpStatus.NOT_FOUND;
```

Comment:
If we fail one one and get another we still have 200? we will only get 404 if we fail on all? We maybe should discus the best way forward for this.

Reply:
This is how the code in subscription looks. I mentioned this to @e-pettersson-ericsson and I some point someone thought this was a good way to do it. I guess it kind of makes sense.

Reply:
Shouldn't we return http error "204 No Content" or something similar when no failed-notifications has been found.
According to some pages I read about 404, it refers to only issues from the httpd web-server part of the application and not failure that happens in the code in the application, e.g. mongodb lookup functionality or what ever executes behind the resource.

If some failed-niotifications is found and some not, then its more a http 200 return code, if you ask me. I guess we return also empty json object for those failed-notifications that was not found.

Reply:
204 is only used if there is no content to send to the user but the headers might be useful. That's not really our case here.

404 is perfectly fine as it basically means "The server can not find requested resource.". But that would only be if we had the endpoint /failed-notifications/sub1,sub2 as the resource "sub1" and "sub2" doesnt exist.

With /failed-notifications?subscriptionName=Sub1,Sub2 i believe 200 is fine as the "query" went ok but we didnt get any data.

Reply:
So we shouldn't return 404 at all then? You can look at the documentation I updated for failed-notifications.md to see an example object. I now put found and not found notifications in arrays so perhaps you are right.

Reply:
I don't return error code anymore if there are no notifications. You simply get an empty array and 200. Technically nothing went wrong.